ledger/participant-integration-api/src/main/scala/platform/apiserver/error/RejectionGenerators.scala

```diff
@@ -0,0 +1,189 @@
+        case _: LfInterpretationError.WronglyTypedContract =>
+          LedgerApiErrors.InterpreterErrors.GenericInterpretationError.Error(renderedMessage)
+        case LfInterpretationError.NonComparableValues =>
+          LedgerApiErrors.InterpreterErrors.GenericInterpretationError.Error(renderedMessage)
+        case _: LfInterpretationError.ContractIdInContractKey =>
+          LedgerApiErrors.InterpreterErrors.GenericInterpretationError.Error(renderedMessage)
+        case LfInterpretationError.ValueExceedsMaxNesting =>
+          LedgerApiErrors.InterpreterErrors.GenericInterpretationError.Error(renderedMessage)
+        case _: LfInterpretationError.ContractIdComparability =>
+          LedgerApiErrors.InterpreterErrors.GenericInterpretationError.Error(renderedMessage)
+      }
+    }
+
+    def processInterpretationError(
+        err: LfError.Interpretation.Error,
+        detailMessage: Option[String],
+    ): BaseError =
+      err match {
+        case Interpretation.Internal(location, message) =>
+          LedgerApiErrors.InternalError.Interpretation(location, message, detailMessage)
+        case m @ Interpretation.DamlException(error) => processDamlException(error, m.message)
+      }
+
+    def processLfError(error: LfError) = {
+      val transformed = error match {
+        case LfError.Package(packageError) => processPackageError(packageError)
+        case LfError.Preprocessing(processingError) => processPreprocessingError(processingError)
+        case LfError.Interpretation(interpretationError, detailMessage) =>
+          processInterpretationError(interpretationError, detailMessage)
+        case LfError.Validation(validationError) => processValidationError(validationError)
+        case e
+            if e.message.contains(
+              "requires authorizers"
+            ) => // Keeping this around as a string match as daml is not yet generating LfError.InterpreterErrors.Validation
+          LedgerApiErrors.InterpreterErrors.AuthorizationError.Reject(e.message)
+      }
+      toGrpc(transformed)
+    }
+
+    val rej = cause match {
+      case ErrorCauseExport.DamlLf(error) => processLfError(error)
+      case x: ErrorCauseExport.LedgerTime =>
+        toGrpc(LedgerApiErrors.CommandPreparation.FailedToDetermineLedgerTime.Reject(x.explain))
+    }
+    Some(rej)
+  }
+
+  def submissionResult(result: Try[state.v2.SubmissionResult]): Option[Try[Unit]] = {
+    result match {
+      case Success(state.v2.SubmissionResult.Acknowledged) => None
+      case Success(grpcError @ state.v2.SubmissionResult.SynchronousError(_)) =>
+        Some(Failure(grpcError.exception))
+      case Failure(_) => None
+    }
+  }
+
+  def validationFailure(reject: StatusRuntimeException)(implicit
+      logger: ContextualizedLogger,
+      loggingContext: LoggingContext,
+      correlationId: CorrelationId,
+  ): StatusRuntimeException = {
+    val description = reject.getStatus.getDescription
```

Comment:
I really would change the validation to return Either.Left with a BaseError instead of converting status runtime exceptions to other status runtime exceptions using string parsing.

Reply:
The original purpose of this class is to convert old types of rejection errors into new error codes rejections. So, if the upstream sends the correct error codes, we can even live without this converted. I'll add a to do

Later edit: I think this is also actually what you meant 👍